### PR TITLE
feat(GROW-2929): update lwgenerate to support GCP generation

### PR DESCRIPTION
**Pull request by leijin-lw**
## How did you test this change?

Add unit tests

### CODE REVIEW

File: lwgenerate/gcp/gcp.go [hunk 1]
```diff
@@ -485,6 +541,7 @@ func createLaceworkProvider(laceworkProfile string) (*hclwrite.Block, error) {
 }
 
 func createGcpProvider(
+	extraProvidedArguments map[string]interface{},
```
Comment (ipcrm) [inline lwgenerate/gcp/gcp.go:544]: nit: Provided -> Provider

File: lwgenerate/gcp/gke.go [hunk 2]
```diff
@@ -33,12 +35,13 @@ func (args *GenerateGkeTfConfigurationArgs) Generate() (string, error) {
...
 	}
 
-	requiredProviders, err := createRequiredProviders(args.UseExistingRequiredProviders)
+	requiredProviders, err := createRequiredProviders(args.UseExistingRequiredProviders, args.ExtraBlocksRootTerraform)
 	if err != nil {
 		return "", errors.Wrap(err, "failed to generate required providers")
 	}
 
 	gcpProvider, err := createGcpProvider(
+		map[string]interface{}{},
```
Comment (ipcrm) [inline lwgenerate/gcp/gke.go:44]: For GKE, we likely need to pass provider args here as well or no?

**Review by ipcrm — COMMENTED** [review]
Looking good! One question on GKE's GCP provider

File: lwgenerate/gcp/gcp.go (hunk 1 above)
Comment (leijin-lw) [reply lwgenerate/gcp/gcp.go]: Done

File: lwgenerate/gcp/gke.go (hunk 2 above)
Comment (leijin-lw) [reply lwgenerate/gcp/gke.go]: Done

**Review by ipcrm — APPROVED** [review]

**Pull request merged**